lib/libutee/tee_api_objects.c

```diff
@@ -583,14 +589,29 @@ TEE_Result TEE_GetNextPersistentObject(TEE_ObjectEnumHandle objectEnumerator,
 {
 	TEE_Result res;
 
-	res =
-	    utee_storage_next_enum(objectEnumerator, objectInfo, objectID,
-				   objectIDLen);
+	if (!objectID) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto err;
+	}
 
-	if (res != TEE_SUCCESS && res != TEE_ERROR_ITEM_NOT_FOUND)
-		TEE_Panic(0);
+	if (!objectIDLen) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto err;
+	}
 
-	return res;
+	res = utee_storage_next_enum(objectEnumerator, objectInfo, objectID,
+				     objectIDLen);
+
+	if (res == TEE_SUCCESS)
+		goto out;
+err:
+	if (res == TEE_ERROR_ITEM_NOT_FOUND ||
```

Comment:
As Jens noted in #355, `If you added res == TEE_SUCCESS here you could get rid of the if just above`